app/controllers/users_controller.rb

```diff
@@ -1,17 +1,20 @@
 class UsersController < ApplicationController
-  before_action :logged_in_user,  only: [:show, :edit, :update, :index]
+  before_action :logged_in_user,  only: [:show, :edit, :update, :index, :destroy]
   before_action :correct_user,    only: [:show, :edit, :update, ]
-  before_action :admin_user,      only: :index
+  before_action :admin_user,      only: [:index, :destroy]
 
   def new
     @user = User.new
   end
+
   def show
     @user = User.find(params[:id])
```

Comment:
`find` method could throw an exception, use `find_by` instead
apply the same way for another place
```suggestion
    @user = User.find_by(id: params[:id])
```